Created functionality for Login Authentiaction

AnaGetovska opened the pull request:
The login authentication logic is structured at AuthContext class and useAuth hook.

superKalo commented on src/AuthContext.jsx:
```diff
@@ -0,0 +1,38 @@
+	)
+}
+
+export { AuthContext, AuthProvider }
```
Maybe move this to the context(s) directory?

superKalo commented on src/components/Login.js:
```diff
@@ -1,11 +1,74 @@
+                    <VStack bg="white" paddingY="5em" marginX="20em" className="form-wrap">
+                        {hasError && <Box>{hasError}</Box>}
+                        <Box>
+                            <Input name="username" placeholder="Потребителско име" required onChange={handleChange} />
```
This should be "факултетен номер", right? Add validation for numbers only.

superKalo commented on src/components/Login.js:
```diff
@@ -1,11 +1,74 @@
+                            <Input name="username" placeholder="Потребителско име" required onChange={handleChange} />
+                        </Box>
+                        <Box>
+                            <Input name="password" placeholder="Парола" type="password" required onChange={handleChange} />
```
One more validation rule needed - should be 4 characters or more (as far as I cross-checked on the WebStudent login form)

superKalo commented on src/components/Login.js:
```diff
@@ -1,11 +1,74 @@
+            <Box bg="purple" paddingY="12em">
+                <form onSubmit={handleSubmit}>
+                    <VStack bg="white" paddingY="5em" marginX="20em" className="form-wrap">
+                        {hasError && <Box>{hasError}</Box>}
```
Please make the error UI look more like an error :)

superKalo commented on src/components/Login.js:
```diff
@@ -1,11 +1,74 @@
-        <>
-            <h1><Link to="dashboard">Login</Link></h1>
-        </>
+        <ChakraProvider>
```
Please wrap <ChakraProvider> in the root of the app. Otherwise - no other components will be able to use Chakra components instead this specific one.

superKalo commented on src/components/Logout.js:
```diff
@@ -1,10 +1,13 @@
         <>
             <h1>Logout</h1>
-            <Link to="../">Login</Link>
+            <Button onClick={removeAuth}>Logout</Button>
```
When I click it once - it takes me to `/logout` route and nothing happens. When I click it a second time - it logs me out.

superKalo commented on src/App.js:
```diff
@@ -1,15 +1,22 @@
+
+	if(!user){
+		return <Login />
+	}
```
This flow works, but it's not optimal and has edge cases. When I log in - the route does not change (from /login to /dashboard for example). When I log out, the route also stays /logout.

Ideally, we need to have [Protecter Routes](https://blog.logrocket.com/complete-guide-authentication-with-react-router-v6/#creating-protected-routes). Then - if someone tries to access /dashboard route and it's not logged in - redirect to /login route.

superKalo commented on package.json:
```diff
@@ -34,10 +39,11 @@
+    "eslint": "^8.7.0",
     "eslint-config-prettier": "^8.6.0",
-    "prettier": "2.8.3"
+    "prettier": "^2.5.1"
```
Why did you change the prettier lib version?

superKalo commented on package.json:
```diff
@@ -34,10 +39,11 @@
+    ]
   },
   "devDependencies": {
+    "eslint": "^8.7.0",
```
Rreate-react-app comes with eslint config. Maybe there is no need to include eslint in our dev dependencies too?

AnaGetovska replied:
In our previous conversation you said that maybe I haven't installed eslint prettier. I double-checked and saw in Prettier ESlint prerequisites that I need to install prettier and eslint packages. [ESlint Prettier ReadMe](https://github.com/idahogurl/vs-code-prettier-eslint/blob/master/README.md). Do you think I should remove it?

superKalo replied:
Yes. These packages (eslint in this case) must be installed either locally or globally. The reate-react-app comes with them as a dep. So no need to manually install them or add them in here. So please remove eslint (`npm uninstall eslint --save-dev`).

AnaGetovska commented:
- Added faculty number validation
- Better error messaging
- Private routing
- Login, Logout, Dashboard redirection added
- Moved useAuth into hooks directory
- Moved AuthContext into context directory

superKalo commented on src/components/Login.js:
```diff
@@ -1,11 +1,123 @@
+	})
+
+	const [hasError, setError] = useState(false)
+	const [isInvalid, setIsInvalid] = useState(false)
```
There is something wrong with the error mechanism - no error gets displayed when I try to log in with wrong credentials.
